iphone/cli/commands/_build.js

```diff
@@ -2265,6 +2270,15 @@ iOSBuilder.prototype.validate = function validate(logger, config, cli) {
 						cli.scanHooks(path.join(module.modulePath, 'hooks'));
 					}, this);
 
+					// Exclude arm64 architecture from simulator build in XCode 12+ - TIMOB-28042
+					if (this.legacyModules.size > 0 && parseFloat(this.xcodeEnv.version) >= 12.0) {
+						if (await processArchitecture() === 'arm64') {
```

Comment:
Just reference `process.arch`. It will be set to `"arm64"` on Apple Silicon. No need to run `exec()` and no need to make this callback async.